File: systems-for-design-and-implementation/labs/lab2/LibraryNetworking/src/library/network/utils/LibraryRPCConcurrentServer.java
package library.network.utils;

import library.network.rpcprotocol.LibraryClientRPCWorker;
import library.services.ILibraryServer;

import java.net.Socket;

public class LibraryRPCConcurrentServer extends AbstractConcurrentServer
{
    private ILibraryServer libraryServer;

    public LibraryRPCConcurrentServer(int port, ILibraryServer libraryServer)
    {
        super(port);
        this.libraryServer = libraryServer;
        System.out.println("LibraryRPCConcurrentServer");
    }

    @Override
    protected Thread createWorker(Socket client)
    {
        LibraryClientRPCWorker worker = new LibraryClientRPCWorker(libraryServer, client);
        return new Thread(worker);
    }
}

File: systems-for-design-and-implementation/labs/lab3/LibraryNetworking/src/library/network/utils/LibraryRPCConcurrentServer.scala
package library.network.utils

import java.net.Socket

import library.network.rpcprotocol.LibraryClientRPCWorker
import library.services.ILibraryServer

class LibraryRPCConcurrentServer(val port: Int, val libraryServer: ILibraryServer)
    extends AbstractConcurrentServer(port)
{
    println("LibraryRPCConcurrentServer")

    protected def createWorker(client: Socket): Thread =
    {
        val worker: LibraryClientRPCWorker = new LibraryClientRPCWorker(libraryServer, client)
        new Thread(worker)
    }
}
